modules/pam_env/pam_env.c

```diff
@@ -212,6 +379,52 @@ _assemble_line(FILE *f, char *buffer, int buf_len)
     return used;
 }
 
+static int read_file(const pam_handle_t *pamh, const char*filename, char ***lines)
+{
+    FILE *conf;
+    char buffer[BUF_SIZE];
+
+    D(("Parsed file name is: %s", filename));
+
+    if ((conf = fopen(filename,"r")) == NULL) {
+      pam_syslog(pamh, LOG_ERR, "Unable to open env file: %s", filename);
+      return PAM_IGNORE;
+    }
+
+    size_t i = 0;
+    *lines = malloc((i + 1)* sizeof(char**));
+    if (lines == NULL) {
```

Comment:
Likewise, if `*lines` has been assigned, then `lines` is definitely non-NULL, you probably meant to check `*lines` instead.

Reply:
Yes, you are right